website/docs/opengl/1-hello-window.md

```diff
@@ -0,0 +1,263 @@
+
+If you've used, or are using, an older version of .NET, you will be familliar with the standard C# entry point. .NET 6 changes this by default, however Silk works best using the old format. If you're using an older version of .NET, you can skip this bit.
+
+Add the following code to your file:
+
+```cs
+// If you're using an earlier version of .NET which doesn't have file-scoped namespaces (.NET 5 or earlier), you'll need to encase your class inside the namespace.
+namespace MySilkProgram;
+
+public class Program
+{
+    public static void Main(string[] args) { }
+}
+```
+
+## Installing Silk.NET
+Before you can get started using Silk.NET, you'll want to install Silk.NET. For the purposes of this tutorial, we're going to assume you're already familliar with creating a C# project, as well as how to install packages with your chosen editor.
+
+While there are many packages available, we'll only need three for this tutorial.
+* [Silk.NET.Windowing](https://www.nuget.org/packages/Silk.NET.Windowing)
+* [Silk.NET.Input](https://www.nuget.org/packages/Silk.NET.Input)
+* [Silk.NET.OpenGL](https://www.nuget.org/packages/Silk.NET.OpenGL)
+
+We won't be using the `OpenGL` package in this tutorial, however we'll be using it in every other tutorial, so it's a good idea to install it now.
+
+Using the dotnet cli, installing the packages is simple:
+
+```
+dotnet add package Silk.NET.Windowing
+dotnet add package Silk.NET.Input
+dotnet add package Silk.NET.OpenGL
+```
+
+## Creating a Window
+The first thing we need before we can even think about rendering some graphics is a window to render to.
+
+In other languages & implementations, you will often have to write a ton of boilerplate code to create the window & render loop, and dealing with the OpenGL context. Fortunately for us, Silk.NET features a built-in easy windowing library that does all this for us!
+
+Add the following using directives to the top of your file:
+
+```cs
+using Silk.NET.Input;
+using Silk.NET.Maths;
+using Silk.NET.Windowing;
+```
+
+### The Window Options
+Before we can create a window, we need to tell it how to behave on startup. This is done with the [WindowOptions](https://github.com/dotnet/Silk.NET/blob/main/src/Windowing/Silk.NET.Windowing.Common/Structs/WindowOptions.cs) struct.
+
+Now, the window options struct contains *a lot* of parameters. Fortunately for us, we don't need to set all these, as the window options struct contains a `Default` property that sets these for us.
+
+In your `Main` method, add the following:
+
+```cs
+WindowOptions options = WindowOptions.Default;
+```
+
+We'd still like to change a few of these options for our application, though, so go ahead and add the following underneath that line:
+
+```cs
+options.Size = new Vector2D<int>(800, 600);
+options.Title = "My first Silk.NET application!";
+```
```

Comment:
I left it out for a couple reasons - first, I was trying to stick as much as possible to the already existing samples (so if I change this, perhaps it should be changed in the sample too?), but I also did it to make it a bit easier to read, so that a newbie can see "okay, we are using the default options and then changing a few bits", I doubt many of them have seen the `with` syntax.. Heck, I didn't know it existed until reading this

Reply:
iirc when I was writing the OpenGL project templates we had the same discussion, and I was on the same spot as you are now of just finding out that syntax existed lol

Back then we decided to use the new syntax, so I think we should stick to that

Reply:
alrighty, i'll update it with the new syntax.